alonzo/formal-spec/introduction.tex

```diff
@@ -1,46 +1,52 @@
+based on e.g. the size of the transaction that includes the script.
 
-In contrast, non-native scripts can perform arbitary
+In contrast, \emph{non-native} scripts can perform arbitrary
 (and, in principle, Turing-complete) computations.
-In order to bound execution costs to a pre-determined constant, we use a standard ``fuel'' approach~\cite{XX}.
+In order to bound execution costs to a pre-determined constant, we use a standard ``fuel'' approach~\cite{XX}.\TODO{And this one, please!}
 We require transactions that use non-native scripts
 to have a budget in terms of a number of abstract $\ExUnits$.
-These provide a quantitative bound on resource usage in terms of metrics such as memory usage or abstract execution steps.
-This abstract budget is then used as part of the concrete transaction fee calculation.
+This budget gives a quantitative bound on resource usage in terms of a number of specific metrics, including memory usage or abstract execution steps.
+The budget is then used as part of the transaction fee calculation.
 
-To change the $\ExUnits$ required to
-run the same computation without a hard fork (for example because a
-more efficient interpreter was introduced), every scripting language
-converts the actual execution cost into $\ExUnits$ using a cost model,
-$\CostMod$, which is a protocol parameter.
+Every scripting language
+converts the calculated execution cost into a number of $\ExUnits$s using a cost model,
+$\CostMod$, which is provided as a protocol parameter.
+This allows execution costs (and so transaction fees) to be varied without requiring a major protocol version change (``hard fork'').
+This may be used, for example, if a more efficient interpreter is produced.
 
 \subsection{Extended UTxO}
 
-The specification of the extended UTxO model follows the description in~\cite{plutus_eutxo}.
-All transaction outputs locked by non-native scripts must include the hash an additional ``data'' item, which the transaction needs to supply. This data could also have been stored directly in the UTxO, but this scheme has been chosen for space efficiency reasons. The data is passed to a script that validates that the output is spent correctly, and can be used to encode state.
-All transactions also need to supply a \emph{redeemer} for everything that is validated by a script, which is an additional piece of data passed to the script. The redeemer can be regarded as user-input for the scripts. Note that the same script can be used for multiple different purposes in the same transaction, so in general it does not suffice to include one redeemer per script.
\ No newline at end of file
+The specification of the extended UTxO model follows the description that was given in~\cite{plutus_eutxo}.
+All transaction outputs that are locked by non-native scripts must include the hash of an additional ``data'' item. The actual data item needs to be supplied in the body of the transaction, and can be used to encode state, for example.
```

Comment:
Should we call these "datum objects" throughout (especially this paragraph) so that it is easier to keep track of what datum objects are and where they appear in tx's and the utxo?

Reply:
I renamed it to "datum", I think calling it "datum object" is a bit too much

Reply:
datum is great. *datum objects* could be useful in case you dont want to pluralize datum as "datums"

Reply:
good idea!